src/EFCore/Metadata/Internal/InternalModelBuilder.cs

```diff
@@ -220,33 +220,87 @@ public InternalModelBuilder([NotNull] Model metadata)
         ///     any release. You should only use it directly in your code with extreme caution and knowing that
         ///     doing so can result in application failures when updating to a new Entity Framework Core release.
         /// </summary>
-        public virtual InternalEntityTypeBuilder AssociationEntity(
+        public virtual InternalEntityTypeBuilder HasAutomaticAssociationEntity(
```

Comment:
```suggestion
        public virtual InternalEntityTypeBuilder HasAssociationEntity(
```
The "Automatic" part is implied

Reply:
Really! 🤨 Obvious to anyone calling that code independent of our conventions?

Reply:
What else could this method do?

Reply:
Create an explicit association entity. Indeed that's what I would assume it would do. You need to know that 2 skip navigations => implicit. Why not just say so?

Reply:
If you were to name a method that sums two numbers which one you'd prefer?
```C#
int Sum(int x, int y)
```
or
```C#
int SumTwoIntegers(int x, int y)
```

Reply:
This is nowhere near that clear.

Reply:
Ok, that wasn't the best example. But to someone who knows what "Automatic" refers to it is clear that's the only thing it can do with the inputs. And to someone who doesn't know it doesn't add any useful information.

Reply:
To someone who doesn't know what "Automatic" refers to it makes you stop and think and go "What does automatic mean? Do I want that?" Maybe that stops you from calling that method and makes you look for other ways of doing what you want to do. If you do know what "Automatic" means then maybe it confirms that this is exactly what you wanted to do and not something else which you don't want (e.g. create an explicit association). And all for the small, small price of one word in a method name on an internal class.

But all of that said, the only reason I had it here was so it could be called from both conventions. Since we now only call it in `M2MConvention` shall we just move the whole method there?

Reply:
Sure, though I'd prefer that the method that calls `AddEntityType` is on `InternalModelBuilder` and exposed through `IConventionModelBuilder.Entity`

Reply:
At this stage I don't need to make it public, but if you think it would be useful to others, sure. All it would do is take 2 entity types, construct a name from their names and call `AddEntityType`. Let's agree on a signature so we don't go through all of this again. How about:

```C#
public IConventionEntityTypeBuilder Entity(IConventionEntityType leftEntity, IConventionEntityType rightEntity)
```

or

```C#
public IConventionEntityTypeBuilder Entity(string leftEntityName, string rightEntityName)
```

Reply:
At this stage I don't need to make it public, but if you think it would be useful to others, sure. 

All it would do is take 2 entity types, construct a name from their names and call `AddEntityType`. I was thinking:

```C#
public IConventionEntityTypeBuilder Entity(IConventionEntityType leftEntity, IConventionEntityType rightEntity)
```

or

```C#
public IConventionEntityTypeBuilder Entity(string leftEntityName, string rightEntityName)
```

What were you thinking of passing in?

Reply:
Oh. I see. The bool is `fromDataAnnotation`, and the `Type` is for `Dictionary<string, object>` and the `string` is the already constructed name. I'm fine with that, but I think that's really something @smitpatel should decide in #9914 . So if it's OK I'll just pull it all into `M2MConvention`  for now and when @smitpatel does that work we can change it to call that method.

Reply:
#9914 will add 
````
        IConventionEntityTypeBuilder Entity([NotNull] string name, [NotNull] Type clrType, bool? shouldBeOwned = false, bool fromDataAnnotation = false);
```

Reply:
Ok

Reply:
If you add something which would be changed when #9914 is finished, just add TODO note with 9914 issue number. I will take care of it.